homeassistant/components/mikrotik/hub.py

```diff
@@ -384,12 +376,23 @@ async def async_setup(self):
         self._mk_data = MikrotikData(self.hass, self.config_entry, api)
         await self.async_add_options()
         await self.hass.async_add_executor_job(self._mk_data.get_hub_details)
-        await self.hass.async_add_executor_job(self._mk_data.update)
+        await self.async_refresh()
 
+        self.config_entry.async_on_unload(
+            self.config_entry.add_update_listener(async_options_updated)
+        )
         self.hass.config_entries.async_setup_platforms(self.config_entry, PLATFORMS)
         return True
 
 
+async def async_options_updated(hass, entry):
+    """Triggered by config entry options updates."""
+    hass.data[DOMAIN][entry.entry_id].update_interval = timedelta(
+        seconds=entry.options[CONF_SCAN_INTERVAL]
+    )
+    await hass.data[DOMAIN][entry.entry_id].async_request_refresh()
```

Comment:
```suggestion
    coordinator: DataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
    coordinator.update_interval = timedelta(seconds=entry.options[CONF_SCAN_INTERVAL])
    await coordinator.async_request_refresh()
```

Reply:
Done, please let me know if this is ok to merge or anything else needs to be checked.